src/libraries/Microsoft.Extensions.Logging.Abstractions/src/BufferedLogRecordMetadata.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+#if NET8_0_OR_GREATER
+using System.Diagnostics;
+#endif
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Holds metadata for a buffered log record.
+    /// </summary>
+    public struct BufferedLogRecordMetadata
+    {
+        /// <summary>
+        /// Gets the time when the log record was recorded.
+        /// </summary>
+        public DateTimeOffset CreationTime { get; set; }
+
+        /// <summary>
+        /// Gets the ID of the thread that created the log record.
+        /// </summary>
+        public int? ManagedThreadId { get; set; }
+
+#if NET8_0_OR_GREATER
```

Comment:
Why is this needed?

Reply:
The ActivitySpanId and ActivityTraceId don't exist in the older TFM that are being built, as far as I can tell.

Reply:
Yeah, I saw that too. But when I build locally, I get errors that these symbols don't exist.

Reply:
try to reference `System.Diagnostics.DiagnosticSource , Version=8.0.0.1` or later version for the targets complains about that.

Reply:
OK, I'm trying that out...